card/mod/machines/lib/javascript/decko/filter.js.coffee

```diff
@@ -1,187 +1,122 @@
-$.extend decko,
-  filterCategorySelected: ($selected_item) ->
-    addFilterDropdown = $selected_item.closest("._add-filter-dropdown")
-    category = $selected_item.data("category")
-    widget = addFilterDropdown.closest("._filter-widget")
-    removeCategoryOption(addFilterDropdown, category)
-    showFilterInputField(category, widget)
+#decko.slotReady (slot) ->
+#  slot.find("._filter-widget").each ->
+#    showDefaultFilters $(this)
+
+decko.slotReady (slot) ->
```

Comment:
Two main changes in this file:
1. moved all the filtered list (filter items) stuff to a separate file.
2. restructured the filter handling into more object oriented structure to support resets and other external filter manipulations.

The reset change involves starting with a full list of clean prototypes. Before there were only prototypes of hidden filters.  Now there's a full list, and we clone and initialize and show the active ones when the page is loaded.